### DISCUSSION TITLE
Select Topic1

### DISCUSSION THREAD

**Discussion by mohammadsedigh** [community/community#123917] [Pull Requests]
### Select Topic Area

Question

### Body

Select Topic Area
Question

Body
Select Topic Area
Product Feedback

Body
We use Github Enterprise at the company where I work and it is common for people to start a review in a Pull Request and the PR is merged before the end of the review because there is no way to signal to the PR author that a review has been started.

Many times the reviewer has already reviewed 90% of the PR but the author merged before the end even though the reviewer had important considerations about the PR.

A feature to notify the PR author about the start of the review would be interesting to avoid this.

**Comment by MohamadSD2002** [comment]
We use GitHub Enterprise at the company where I work, and it is common for people to start a review in a Pull Request (PR). However, sometimes the PR is merged before the review is completed because there is no way to signal to the PR author that a review has been started.

Often, the reviewer has already reviewed 90% of the PR, but the author merges it prematurely, even though the reviewer had important considerations about the PR.

A feature to notify the PR author about the start of the review would be beneficial to avoid this issue. Here are a few suggestions to improve the review process:

Draft Reviews: GitHub could implement a feature where starting a review marks the PR as "Under Review" or "In Progress." This would make it clear to the author that a review is ongoing.

Review Status Indicator: Adding a status indicator on the PR page that shows which reviewers have started their review and how much progress they have made can be very helpful.

Notifications: Automatic notifications to the PR author when a reviewer starts a review can prevent premature merging.

Review Check Requirement: Make it a requirement that all reviews must be either approved or explicitly requested for changes before the PR can be merged.

Custom Labels: As a temporary solution, consider using custom labels like "Review in Progress" to manually indicate when a review has started.

Implementing these features or processes would help ensure that all reviews are completed thoroughly and that important feedback is not overlooked.

Thank you for considering these suggestions!

If u liked this answer, u can accept it :)
